src/java/org/apache/cassandra/concurrent/TaskFactory.java

```diff
@@ -128,7 +128,41 @@ protected LocalAware() {}
         public Runnable toExecute(Runnable runnable)
         {
             // no reason to propagate exception when it is inaccessible to caller
-            return ExecutionFailure.suppressing(ExecutorLocals.propagate(), runnable);
+            Runnable suppressing = ExecutionFailure.suppressing(ExecutorLocals.propagate(), runnable);
+            
+            if (runnable instanceof DebuggableTask)
+            {
+                return new DebuggableTask.RunnableDebuggableTask()
+                {
+                    private final DebuggableTask debuggable = (DebuggableTask) runnable;
+                    
+                    @Override
+                    public void run()
+                    {
+                        suppressing.run();
+                    }
+
+                    @Override
+                    public long creationTimeNanos()
+                    {
+                        return debuggable.creationTimeNanos();
+                    }
+
+                    @Override
+                    public long startTimeNanos()
+                    {
+                        return debuggable.startTimeNanos();
+                    }
+
+                    @Override
+                    public String description()
+                    {
+                        return debuggable.description();
+                    }
+                };
+            }
```

Comment:
Before the simulator changes, this bit wouldn't have been necessary, as everything got wrapped up in `FutureTask`. Is there a better way?

CC @belliottsmith

Reply:
It's similar to what we still do w/ `FutureTask` for NTR stage requests (and did for everything in the old patch).